### PR TITLE
youtube

### PR DESCRIPTION
**Pull request by dleng16**
youtube v3 api

endpoint(s): 
/youtube/searchList 

sample postman request:
http://localhost:8080/youtube/searchList?part=snippet&maxResults=5&q=COVID-19%20news

### CODE REVIEW

File: apis/youtubeApi.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
+const tokens = require('../config/tokens');
+const request = require('request');
+const Youtube = require("youtube-api");
+
+Youtube.authenticate({
+    type: "key",
+    key: tokens.youtube.key
```
Comment (jchen006) [inline apis/youtubeApi.js:7]: can you pass the keys to everyone some time later?

File: apis/youtubeApi.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    try {
+        const response = await Youtube.search.list(query);
+        console.log(response);
+        return response;
+    } catch (error) {
+        throw new Error(error);
+        return;
+    }
+};
+// const query = {
```
Comment (jchen006) [inline apis/youtubeApi.js:20]: remove this

File: controllers/youtube.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+const { youtubeMapper } = require('../mappers/youtubeMapper');
+
+const getSearchList = async (req, res) => {
+    let qs = req.query;
+    const response = await querySearchList(qs);
+    const { data, status } = response;
+    if (status == 200 && data) {
+        let items = data.items;
+        const mappedSearch = items.map(items => youtubeMapper(items));
+        console.log(mappedSearch);
```
Comment (jchen006) [inline controllers/youtube.js:11]: remove this

File: controllers/youtube.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
+const { querySearchList } = require('../apis/youtubeApi');
+const { youtubeMapper } = require('../mappers/youtubeMapper');
+
+const getSearchList = async (req, res) => {
+    let qs = req.query;
+    const response = await querySearchList(qs);
+    const { data, status } = response;
+    if (status == 200 && data) {
+        let items = data.items;
```
Comment (jchen006) [inline controllers/youtube.js:9]: you could write it like this `const { items=[] } = data;` basically the `=[]` acts as a default like python if there is no items

File: apis/youtubeApi.js (hunk 1 above)
Comment (dleng16) [reply apis/youtubeApi.js:7]: sure

File: apis/youtubeApi.js (hunk 2 above)
Comment (dleng16) [reply apis/youtubeApi.js:20]: got it

File: routes/youtube.js [hunk 5]
```diff
@@ -0,0 +1,9 @@
+const express = require('express');
+const router = express.Router();
+const {
+    getSearchList
+} = require('../controllers/youtube')
+
+router.get('/searchList', getSearchList)
```
Comment (jchen006) [inline routes/youtube.js:7]: I know you raised this to me but I would like to rename this to just `search`

**Review by jchen006 — CHANGES_REQUESTED** [review]
Looks good overall, minor changes needed

File: controllers/youtube.js (hunk 3 above)
Comment (dleng16) [reply controllers/youtube.js:11]: oops

File: controllers/youtube.js (hunk 4 above)
Comment (dleng16) [reply controllers/youtube.js:9]: nice ok

File: routes/youtube.js (hunk 5 above)
Comment (dleng16) [reply routes/youtube.js:7]: ur wish is my cmd

**Review by jchen006 — APPROVED** [review]

**Pull request merged**